doc/contributing/submitting-changes.chapter.md

```diff
@@ -82,6 +82,48 @@ If a security fix applies to both master and a stable release then, similar to r
 
 Critical security fixes may by-pass the staging branches and be delivered directly to release branches such as `master` and `release-*`.
 
+## Deprecating/removing packages {#submitting-changes-deprecating-packages}
+
+When software projects get discontinued or are unmaintained and break, we have to remove them. Another reason would be that there is no maintainer for the package in nixpkgs. In that case, try to find a new maintainer first.
+
+### Steps to remove a package from nixpkgs
+
+We use jbidwatcher as an example for a discontinued project here.
+
+1. have nixpkgs checked out locally and up to date
+1. create a new branch for your change, e.g. `git checkout -b jbidwatcher`
+1. remove the actual package including it's directory, e.g. `rm -rf pkgs/applications/misc/jbidwatcher`
+1. remove the package from list of all packages (`pkgs/top-level/all-packages.nix`)
+1. add error message for package name in `pkgs/top-level/aliases.nix`
+
+    For example in this case:
+    ```
+    jbidwatcher = throw "jbidwatcher was discontinued in march 2021"; # added 2021-03-15
+    ```
+
+    Explain shortly why the package was removed for users that still have it installed.
+
+1. test if the change introduced any issues with `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
```

Comment:
``nix-env -qaP -f . --show-trace`` with aliases off is enough. Not sure how to pass ``config.allowAliases = false;`` to it.

Reply:
Yeah, that's better. nixpkgs-review does not even work in this case :grimacing: 

https://github.com/Mic92/nixpkgs-review/issues/189

Reply:
I run it locally with that option since a bit but taking a look at the issue.